<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">
  <head>
    <meta http-equiv="content-type" content="text/html; charset=utf-8" />
    <meta name="author" content="wayne" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0"/>
    
    <title>fossil HTTPS patch</title>
    
    <link rel="shortcut icon" href="/favicon.ico">
    <link href="/atom.xml" rel="alternate" title="rusty shutter" type="application/atom+xml" />
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/font-awesome/4.2.0/css/font-awesome.min.css">
    <link rel="stylesheet" href="/media/css/style.css">
    <script type="text/javascript" src="//cdnjs.cloudflare.com/ajax/libs/retina.js/1.3.0/retina.min.js"></script>
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/8.4/styles/github.min.css">
    <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/8.4/highlight.min.js"></script>
    <script type="text/javascript"> hljs.initHighlightingOnLoad(); </script>
  </head>
  <body>
      <div id="main" role="main">
        <header>
          <div id="header">
            <h1><a title="rusty shutter" class="" href="/">rusty shutter</a></h1>
          </div>
          <nav>
            
            <span><a title="Archive" href="/archive.html"><i class="fa fa-list-ul"></i></a></span>
            
            <span><a title="Tags" href="/tags.html"><i class="fa fa-tags"></i></a></span>
            
            <span><a title="About" href="/about.html"><i class="fa fa-user"></i></a></span>
            
            <span><a title="Gallery" href="http://foto.lhzhang.com"><i class="fa fa-film"></i></a></span>
            
            <span><a title="Subscribe" href="/atom.xml"><i class="fa fa-rss"></i></a></span>
            
          </nav>
        </header>
        <div id="content">
        <article>
  <section class="title">
    <h2>fossil HTTPS patch </h2>
  </section>
  <section class="meta">
  <span class="time">
    <time datetime="2010-07-30">2010-07-30</time>
  </span>
  
  <span class="tags">
    
    <a href="/tags.html#fossil" title="fossil">#fossil</a>
    
  </span>
  <!-- BEGIN this would not work on any other domain -->
  <span
    class           = "like-wrapper"
    like-shortname  = 'gopherwood'
    like-identifier = 'urn:uuid:e14a2804-f4f2-4037-b1b2-72129894167d'
    like-name       = 'fossil HTTPS patch'
    like-btn        = '&#xf087;'
    like-link       = 'http://lhzhang.com/2010/07/30/patch-for-fossil-scm.html'
    ></span>
  <script type="text/javascript">
    var l = document.createElement('script');
    l.type = 'text/javascript'; l.async = true; l.src = 'http://www.like-btn.com/javascript/widget.js';
    (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(l);
  </script>
  <!-- END this would not work on any other domain -->
  
  </section>
  <section class="post">
  <p>最近用fossil搭了个wiki，不过原版尽管支持https schema的url生成，却漏掉了https header的判断，小小patch了一下</p>

<pre><code>--- src/cgi.c   2010-07-10 01:57:40.000000000 +0900
+++ src/cgi.c   2010-07-21 16:22:41.000000000 +0900
@@ -1140,6 +1140,8 @@
       cgi_setenv("HTTP_IF_NONE_MATCH", zVal);
     }else if( strcmp(zFieldName,"if-modified-since:")==0 ){
       cgi_setenv("HTTP_IF_MODIFIED_SINCE", zVal);
+    }else if( strcmp(zFieldName,"https:")==0 ){
+      cgi_setenv("HTTPS", zVal);
     }
   }
</code></pre>

<p>至于fossil是个什么东西以后有空整理一下，感兴趣的童鞋可以去 <a href="http://www.fossil-scm.org">Fossil</a> 看下</p>

  </section>
  
  <div class="divider">
    <span>
    
    <a href="/2010/07/19/iphone-4-antenna.html"><i class="fa fa-chevron-left"></i></a>
    
    </span>
    <!-- BEGIN comment icon
    <span><a href="javascript:leave_comment();" id="leave_comment_link"><i class="fa fa-comment-o"></i></a></span>
    <span><a href="javascript:collapse_comment();" id="collapse_comment_link" style="display:none;"><i class="fa fa-chevron-up"></i></a></span>
      END comment icon -->

    <span>
    
    <a href="/2010/08/03/feedburner-domain-bind-with-nginx.html"><i class="fa fa-chevron-right"></i></a>
    
    </span>
  </div>

  <!-- BEGIN comment
  <section class="comment">
  <div id="disqus_thread"></div>
  <script type="text/javascript">
      var disqus_shortname = 'gopherwood';
      var disqus_identifier = 'urn:uuid:e14a2804-f4f2-4037-b1b2-72129894167d';

      function leave_comment() {
          var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
          dsq.src = 'http://' + disqus_shortname + '.disqus.com/embed.js';
          (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
          $("#leave_comment_link").css("display", "none");
          $("#collapse_comment_link").css("display", "");
      };
      function collapse_comment() {
          // document.getElementById("disqus_thread").innerHTML = '';
          $("#disqus_thread").slideUp(400, function() {
              $("#disqus_thread").empty();
              $("#leave_comment_link").css("display", "");
              $("#collapse_comment_link").css("display", "none");
              $("#disqus_thread").css("display", "");
          });
      };
  </script>
  <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
  </section>
  END comment -->
  
</article>

        </div>
        <footer>
          <div>
            
            &copy; 2006 ~ 2015 wayne | powered by jekyll | themed by <a href="http://lhzhang.com" title="sext vi">sext vi</a> | fork <a href="https://github.com/waynezhang/blog" title="fork me">me</a>
          </div>
        </footer>
      </div> <!-- main -->
  </body>
</html>
